Library/Homebrew/cli_parser.rb

```diff
@@ -0,0 +1,271 @@
+module HomebrewCli
```

Comment:
I think `HomebrewCLI` or, even better `Homebrew::CLI` reads nicer and is more Ruby-ish.